go/engine/selfprovision.go

```diff
@@ -95,19 +88,16 @@ func (e *SelfProvisionEngine) Run(m libkb.MetaContext) (err error) {
 	e.ekReboxer = newEphemeralKeyReboxer(m)
 
 	// Make new device keys and sign them with current device keys
-	if err = e.provision(m, keys); err != nil {
+	if err := e.provision(m, keys); err != nil {
 		return err
 	}
 
-	// commit the config changes
-	// NOTE we should wrap the Active device change in a transaction as well.
-	if err := tx.Commit(); err != nil {
+	// Finish provisoning by calling SetActiveDevice which sets
```

Comment:
can you put a similar comment for the other callers? just to be explicit so anyone looking in the future understands the reasoning. maybe add something like `This is not revertable, so anything else that fails the provisioning process is best effort since we have commit to this new device`